libc/signal.c

```diff
@@ -0,0 +1,196 @@
+
+/* Typical error message does not exceed ~60 characters, that's why we expect a maximum value a little bit bigger */
+#define MAX_LEN_STRING 100
+#define PATH           "stderr.log"
+
+static FILE *err_file;
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-const-variable"
+static const int signal_codes[] = { SIGABRT, SIGALRM, SIGBUS, SIGCHLD, SIGCONT, SIGFPE, SIGHUP, SIGILL, SIGINT,
+	SIGKILL, SIGPIPE, SIGQUIT, SIGSEGV, SIGSTOP, SIGTERM, SIGTSTP, SIGTTIN, SIGTTOU, SIGUSR1, SIGUSR2, SIGPROF,
+	SIGSYS, SIGTRAP, SIGURG, SIGVTALRM, SIGXCPU, SIGXFSZ };
+
+static const unsigned int signal_codes_len = sizeof(signal_codes) / sizeof(signal_codes[0]);
+#pragma GCC diagnostic pop
+
+TEST_GROUP(signal_psignal);
+
+TEST_SETUP(signal_psignal)
+{
+	err_file = fopen(PATH, "a+");
+}
+
+
+TEST_TEAR_DOWN(signal_psignal)
+{
+	remove(PATH);
+}
+
+
+TEST(signal_psignal, basic)
+{
+/* Disabled because of #695 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/695 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#695 issue");
+#else
+	/* Since these function shall not return a value, we check only errno */
+	char buff[MAX_LEN_STRING];
+
+	for (int i = 0; i < signal_codes_len; i++) {
+		rewind(err_file);
+		memset(buff, 0, sizeof(buff));
+
+		TEST_ASSERT_NOT_NULL(freopen("stderr.log", "w", stderr));
+
+		errno = 0;
+		psignal(signal_codes[i], "Lorem Ipsum");
+		TEST_ASSERT_EQUAL_INT(0, errno);
+
+		fflush(stderr);
+
+		TEST_ASSERT_NOT_NULL(fgets(buff, MAX_LEN_STRING, err_file));
+		TEST_ASSERT_NOT_NULL(strstr(buff, "Lorem Ipsum"));
+		fflush(err_file);
+	}
+	fclose(err_file);
+#endif
+}
+
+
+TEST(signal_psignal, ascii_string)
+{
+/* Disabled because of #695 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/695 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#695 issue");
+#else
+	char buff[MAX_LEN_STRING];
+
+	char ascii_str[2 * MAX_LEN_STRING];
+
+	for (int i = 0; i < 128; i++) {
+		ascii_str[i] = i;
+	}
+	ascii_str[128] = '\0';
+
+	for (int i = 0; i < signal_codes_len; i++) {
+		rewind(err_file);
+		memset(buff, 0, sizeof(buff));
+
+		TEST_ASSERT_NOT_NULL(freopen("stderr.log", "w", stderr));
+
+		errno = 0;
+		psignal(signal_codes[i], ascii_str);
+		TEST_ASSERT_EQUAL_INT(0, errno);
+
+		fflush(stderr);
+
+		TEST_ASSERT_NOT_NULL(fgets(buff, MAX_LEN_STRING, err_file));
+		TEST_ASSERT_NOT_NULL(strstr(buff, ascii_str));
+		fflush(err_file);
+	}
+	fclose(err_file);
+#endif
+}
+
+
+TEST(signal_psignal, psignal_strsignal)
+{
+/* Disabled because of #695 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/695 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#695 issue");
+#else
+	char buff[MAX_LEN_STRING];
+
+	for (int i = 0; i < signal_codes_len; i++) {
+		int length;
+		char *signal_output = strsignal(signal_codes[i]);
+
+		rewind(err_file);
+		memset(buff, 0, sizeof(buff));
+
+		TEST_ASSERT_NOT_NULL(freopen("stderr.log", "w", stderr));
+
+		errno = 0;
+		psignal(signal_codes[i], "Simple Message");
```

Comment:
maybe make it a define instead of repeating it 3 times?

eg. 
```c
#define SIMPLE_MSG "Simple Message"

psignal(signal_codes[i], SIMPLE_MSG)
length = strlen(signal_output) + strlen(SIMPLE_MSG ": ") + 1;  // TODO: explain why + 1
```